operator/apis/resources/v2/user_types.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/gravitational/teleport/api/types"
+)
+
+const DescriptionKey = "description"
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// UserSpec defines the desired state of User
+type UserSpec types.UserSpecV2
```

Comment:
> I do wonder if reusing the types.*Spec for the Kubernetes resource specs is ever going to cause us pain here. 

It will :sweat_smile: 
We have another project where we re-use the entire `types.*Spec`: Terraform Provider for Teleport.
https://github.com/gravitational/teleport-plugins/tree/master/terraform

It's not always pretty but works 99% of the time and saves us tons of development time.
When it doesn't work, we usually need to implement some custom type or some exception, but so far is proving to be worthwhile.

I do agree with you, if it stops working we can always be explicit about the spec and matching JSON Keys.

Reply:
Sounds good, wasn't aware we had prior art on this. I haven't had a play with our Terraform provider yet :')